insteon_mqtt/device/Dimmer.py

```diff
@@ -465,6 +465,69 @@ def set_backlight(self, level, on_done=None):
         msg_handler = handler.StandardCmd(msg, self.handle_backlight, on_done)
         self.send(msg, msg_handler)
 
+    #-----------------------------------------------------------------------
+    def get_flags(self, on_done=None):
+        """Hijack base get_flags to inject extended flags request.
+
+        The flags will be passed to the on_done callback as the data field.
+        Derived types may do something with the flags by override the
+        handle_flags method.
+
+        Args:
+          on_done: Finished callback.  This is called when the command has
+                   completed.  Signature is: on_done(success, msg, data)
+        """
+        seq = CommandSeq(self, "Dimmer get_flags complete", on_done,
+                         name="GetFlags")
+        seq.add(super().get_flags)
+        seq.add(self._get_ext_flags)
+        seq.run()
+
+    #-----------------------------------------------------------------------
+    def _get_ext_flags(self, on_done=None):
+        """Get the Insteon operational extended flags field from the device.
+
+        For the dimmer device, the flags include on-level and ramp-rate.
+
+        Args:
+          on_done: Finished callback.  This is called when the command has
+                   completed.  Signature is: on_done(success, msg, data)
+        """
+        LOG.info("Dimmer %s cmd: get extended operation flags", self.label)
+
+        # Requesting data d1 = 0x01
+        data = bytes([0x01] + [0x00] * 13)
+
+        msg = Msg.OutExtended.direct(self.addr, Msg.CmdType.EXTENDED_SET_GET,
+                                     0x00, data)
+        msg_handler = handler.ExtendedCmdResponse(msg, self.handle_ext_flags,
+                                                  on_done)
+        self.send(msg, msg_handler)
+
+    #-----------------------------------------------------------------------
+    def handle_ext_flags(self, msg, on_done):
+        """Handle replies to the _get_ext_flags command.
+
+        Extended message payload is:
+          D8 = on-level
+          D9 = ramp-rate
+
+        Args:
+          msg (message.InpExtended):  The message reply.
+          on_done:  Finished callback.  This is called when the command has
+                    completed.  Signature is: on_done(success, msg, data)
+        """
+        on_level = msg.data[7]
+        self.db.set_meta('on_level', on_level)
+        ramp_rate = msg.data[8]
```

Comment:
I think ramp rate may be in D7 for dimmers, too.  This is what I get for one of my dimmers:

> 2020-12-30 01:04:08 INFO Protocol: Read 0x51: Ext: 11.22.33->aa.bb.cc Type.DIRECT ext cmd: 2e 00
01 01 00 00 20 20 1b b3 40 00 01 00 00 00
2020-12-30 01:04:08 DEBUG MsgHistory: Received 1 hops, total 2 for 3 entries
2020-12-30 01:04:08 DEBUG Protocol: Passing msg to write handler: ExtendedCmdResponse handler
2020-12-30 01:04:08 UI Dimmer: Dimmer 11.22.33 (kitchen cabinet lights) on_level: 70.19607843137256% ramp rate: 64s

Yet, the actual ramp rate for this dimmer is 2 seconds (0x1b).

Reply:
Oops, you are right.  It looks like D9 is the LED light level.